File: server/data/results/translations/lite/anchored/javascript/bubble_sort.js
function bubble_sort(values) {
    let items = Array.from(values);
    let n = items.length;
    for (let i = 0; i < n; i++) {
        let swapped = false;
        for (let j = 0; j < n - i - 1; j++) {
            if (items[j] > items[j + 1]) {
                let temp = items[j];
                items[j] = items[j + 1];
                items[j + 1] = temp;
                swapped = true;
            }
        }
        if (!swapped) {
            break;
        }
    }
    return items;
}

function main() {
    let data = [5, 2, 9, 1, 5, 6];
    let result = bubble_sort(data);
    let line = "";
    for (let value of result) {
        line = line + String(value) + " ";
    }
    console.log(line.trim());
}

main();

File: server/data/results/translations/lite/anchored/java/bubble_sort.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Writing {

    public static List<Integer> bubble_sort(List<Integer> values) {
        List<Integer> items = new ArrayList<>(values);
        int n = items.size();
        for (int i = 0; i < n; i++) {
            boolean swapped = false;
            for (int j = 0; j < n - i - 1; j++) {
                if (items.get(j) > items.get(j + 1)) {
                    int temp = items.get(j);
                    items.set(j, items.get(j + 1));
                    items.set(j + 1, temp);
                    swapped = true;
                }
            }
            if (!swapped) {
                break;
            }
        }
        return items;
    }

    public static void main(String[] args) {
        List<Integer> data = Arrays.asList(5, 2, 9, 1, 5, 6);
        List<Integer> result = bubble_sort(data);
        String line = "";
        for (Integer value : result) {
            line = line + String.valueOf(value) + " ";
        }
        System.out.println(line.trim());
    }
}